ot/mpz-ot/src/actor/kos/receiver.rs

```diff
@@ -0,0 +1,346 @@
+    BaseOT: OTSetup + VerifiableOTSender<bool, [Block; 2]> + ProtocolMessage + Send,
+    Si: IoSink<Message<BaseOT::Msg>> + Send + Unpin,
+    St: IoStream<Message<BaseOT::Msg>> + Send + Unpin,
+{
+    /// Create a new receiver actor.
+    pub fn new(receiver: Receiver<BaseOT>, sink: Si, stream: St) -> Self {
+        let (command_sender, commands) = mpsc::unbounded();
+
+        Self {
+            receiver,
+            sink,
+            stream: stream.fuse(),
+            state: Default::default(),
+            command_sender,
+            commands,
+        }
+    }
+
+    /// Sets up the receiver with the given number of OTs.
+    pub async fn setup(&mut self, count: usize) -> Result<(), ReceiverActorError> {
+        let mut sink = into_kos_sink(&mut self.sink);
+        let mut stream = into_kos_stream(&mut self.stream);
+
+        self.receiver.setup(&mut sink, &mut stream).await?;
+        self.receiver.extend(&mut sink, &mut stream, count).await?;
+
+        Ok(())
+    }
+
+    /// Returns a `SharedReceiver` which implements `Clone`.
+    pub fn receiver(&self) -> SharedReceiver {
+        SharedReceiver {
+            sender: self.command_sender.clone(),
+        }
+    }
+
+    /// Run the receiver actor.
+    pub async fn run(&mut self) -> Result<(), ReceiverActorError> {
+        loop {
+            futures::select! {
+                msg = self.stream.select_next_some() => self.handle_msg(msg?).await?,
+                cmd = self.commands.select_next_some() => {
+                    if let Command::Shutdown(Shutdown { caller_response }) = cmd {
+                        _ = caller_response.send(Ok(()));
+                        return Ok(());
+                    }
+
+                    self.handle_cmd(cmd).await?
+                },
+            }
+        }
+    }
+
+    async fn start_transfer(
+        &mut self,
+        id: &str,
+        choices: &[bool],
+    ) -> Result<ReceiverKeys, ReceiverError> {
+        let mut keys = self
+            .receiver
+            .state_mut()
+            .as_extension_mut()?
+            .keys(choices.len())?;
+
+        let derandomize = keys.derandomize(choices)?;
+
+        self.sink
+            .send(
+                ActorMessage::TransferRequest(TransferRequest {
+                    id: id.to_string(),
+                    derandomize,
+                })
+                .into(),
+            )
+            .await?;
+
+        Ok(keys)
+    }
+
+    async fn start_verification(&mut self) -> Result<(), ReceiverError> {
+        self.receiver
+            .verify_delta(
+                &mut into_kos_sink(&mut self.sink),
+                &mut into_kos_stream(&mut self.stream),
+            )
+            .await?;
+
+        // Process backlog of pending verifications.
+        let backlog = std::mem::take(&mut self.state.pending_verify);
+        for verify in backlog {
+            self.handle_verify(verify)
+        }
+
+        Ok(())
+    }
+
```

Comment:
/// Handles the `Verify` command from the controller.
///
/// Sends back the [`PayloadRecord`] which the controller must verify itself.